taskcluster/translations_taskgraph/transforms/training_continuation.py

```diff
@@ -0,0 +1,68 @@
+from taskgraph.transforms.base import TransformSequence
+from urllib.parse import urljoin
+import os
+
+MODEL_TRAINING_ARTIFACT_NAMES = (
+    "devset.out",
+    "model.npz",
+    "model.npz.best-bleu-detok.npz",
+    "model.npz.best-bleu-detok.npz.decoder.yml",
+    "model.npz.best-ce-mean-words.npz",
+    "model.npz.best-ce-mean-words.npz.decoder.yml",
+    "model.npz.best-chrf.npz",
+    "model.npz.best-chrf.npz.decoder.yml",
+    "model.npz.decoder.yml",
+    "model.npz.optimizer.npz",
+    "model.npz.progress.yml",
+    "model.npz.yml",
+    "train.log",
+    "valid.log",
+    "vocab.spm",
+)
+
+
+def get_artifact_mount(url, directory, artifact_name):
+    normalized_url = f"{url}/" if not url.endswith("/") else url
+    artifact_url = urljoin(normalized_url, artifact_name)
+    return {
+        "content": {
+            "url": artifact_url,
+        },
+        "file": os.path.join(directory, artifact_name),
+    }
+
+
+def get_artifact_mounts(urls, directory, artifact_names):
+    ret = []
+    for url in urls:
+        artifact_mounts = []
+        for artifact_name in artifact_names:
+            artifact_mounts.append(get_artifact_mount(url, directory, artifact_name))
+        ret.append(artifact_mounts)
+    return ret
+
+
+transforms = TransformSequence()
+
+
+@transforms.add
+def training_continuation(config, jobs):
+    pretrained_models = config.params["training_config"]["experiment"].get("pretrained-models", {})
+    if not any(pretrained_models):
+        for job in jobs:
+            yield job
+    else:
+        model_config = pretrained_models.get("teacher-base")
```

Comment:
This transform's name (`training_continuation`) suggests that it could be used on any training step, but some of the details such as this make it clear it only works for `train-teacher` right now. It would be lovely if it could be made generic enough to work with any training step (eg: by parameterizing some things). If not, it ought to be renamed.

(It can probably be refactored in such a way that most of the code can be re-used even if we maintain separate transforms for each training step.)